databricks-sdk-java/src/main/java/com/databricks/sdk/core/error/AbstractErrorMapper.java

```diff
@@ -0,0 +1,37 @@
+package com.databricks.sdk.core.error;
+
+import com.databricks.sdk.core.DatabricksError;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+abstract class AbstractErrorMapper {
+  @FunctionalInterface
+  protected interface Rule {
+    DatabricksError create(String message, List<ErrorDetail> details);
+  }
+
+  public DatabricksError apply(int code, ApiErrorBody errorBody) {
+    String message = errorBody.getMessage();
+    String errorCode = errorBody.getErrorCode();
+    List<ErrorDetail> details = errorBody.getErrorDetails();
+    if (statusCodeMapping.containsKey(code)) {
+      return statusCodeMapping.get(code).create(message, details);
+    }
+    if (errorCodeMapping.containsKey(errorCode)) {
+      return errorCodeMapping.get(errorCode).create(message, details);
+    }
```

Comment:
I think this is in the wrong order: we should check for specific error code first and then fall back to status code.